Userland/Libraries/LibAudio/Sample.h

```diff
@@ -9,16 +9,80 @@
 #pragma once
 
 #include <AK/Math.h>
+#include <math.h>
 
 namespace Audio {
 using namespace AK::Exponentials;
-// Constants for logarithmic volume. See Sample::linear_to_log
-// Corresponds to 60dB
-constexpr double DYNAMIC_RANGE = 1000;
-constexpr double DYNAMIC_RANGE_DB = 60;
+
+// Logarithmic scaling, as audio should ALWAYS do.
+// Reference: https://www.dr-lex.be/info-stuff/volumecontrols.html
+// We use the curve `factor = a * exp(b * change)`,
+// where change is the input fraction we want to change by,
+// The dynamic range is based on the bit depth which is currently set to
+// 16 bit. This might change in the future and can be changed later.
+// This dynamic range gives us ~96.3 dB of dB range.
+// Reference: https://en.wikipedia.org/wiki/DBFS
+
+constexpr double BIT_DEPTH = 16;
+constexpr double DYNAMIC_RANGE_DB = 20.0 * log10(AK::pow(2.0, BIT_DEPTH));
+constexpr double DYNAMIC_RANGE = AK::pow(10.0, DYNAMIC_RANGE_DB * 0.05);
 constexpr double VOLUME_A = 1 / DYNAMIC_RANGE;
 double const VOLUME_B = log(DYNAMIC_RANGE);
 
+// Format ranges:
+// - Linear:          0.0 to 1.0
+// - Amplitude:       0.0 to 1.0
+// - dB(linear):    ~-96.3 to 0.0
+// - dB(amplitude):  -inf to 0.0
+// - Panning:        -1.0 to 1.0 (Left to Right)
+
+ALWAYS_INLINE double linear_to_amplitude(double const value)
+{
+    // Linear slope at the lower values to avoid asymptotic behaviour
+    // NOTE: These values are dependent on the dynamic range and thus the bit depth.
+    if (value < 0.1)
+        return value * 0.00045;
+
+    return VOLUME_A * exp(VOLUME_B * value);
+}
+
+ALWAYS_INLINE double amplitude_to_linear(double const amplitude)
+{
+    // Linear slope at the lower values to avoid asymptotic behaviour
+    // NOTE: These values are dependent on the dynamic range and thus the bit depth.
+    if (amplitude < 0.00004)
+        return amplitude * 2160;
+
+    return log(amplitude / VOLUME_A) / VOLUME_B;
+}
+
+ALWAYS_INLINE double db_to_linear(double const dB)
```

Comment:
I think this is a misnomer. We're not converting decibels to a linear scale (or the other way around), since decibels are *not* linear. This is converting a decibel value into a 0-1 range where 0 is the lowest representable relative decibel value (~-96dB) due to bitrate, and 1 is the maximum representable relative decibel value (0dB, no headroom).

Reply:
Yes, I see what you mean. The range and headroom must be passed as well for it to make sense.